filemanager/gcsmanager.go

```diff
@@ -85,18 +101,31 @@ func (m *gcsManager) Upload(ctx context.Context, file *os.File, prefixes ...stri
 	ctx, cancel := context.WithTimeout(ctx, m.getTimeout())
 	defer cancel()
 
-	obj := client.Bucket(m.config.Bucket).Object(fileName)
-	w := obj.NewWriter(ctx)
-	if _, err := io.Copy(w, file); err != nil {
-		err = fmt.Errorf("copying file to GCS: %v", err)
-		if closeErr := w.Close(); closeErr != nil {
-			return UploadedFile{}, fmt.Errorf("closing writer: %q, while: %w", closeErr, err)
-		}
+	object := client.Bucket(m.config.Bucket).Object(fileName)
+	if m.config.doNotOverWrite {
+		object = object.If(storage.Conditions{DoesNotExist: true})
+	}
+	w := object.NewWriter(ctx)
 
-		return UploadedFile{}, err
+	if _, err := io.Copy(w, file); err != nil {
+		return UploadedFile{}, fmt.Errorf("copying file to writer: %v", err)
```

Comment:
We can probably wrap the error.
```suggestion
		return UploadedFile{}, fmt.Errorf("copying file to writer: %w", err)
```